From 167af26fc9f7c3db14a20c6a732b97107060def9 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Tue, 26 Jun 2012 17:47:16 +0800
Subject: ENGR00213014-4 HDMI HDCP: IOMUX define for MX6X

Added IOMUX and pad setting for HDMI DDC for mx6q/mx6dl.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 arch/arm/plat-mxc/include/mach/iomux-mx6dl.h |   12 ++++++++----
 arch/arm/plat-mxc/include/mach/iomux-mx6q.h  |    8 ++++----
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h b/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
index 1fa0e82..4d0fa92 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mx6dl.h
@@ -2665,11 +2665,13 @@
 #define MX6DL_PAD_KEY_COL3__ENET_CRS                                           \
 		IOMUX_PAD(0x0638, 0x0250, 1, 0x0000, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_COL3__HDMI_TX_DDC_SCL                                    \
-		IOMUX_PAD(0x0638, 0x0250, 2, 0x0860, 1, NO_PAD_CTRL)
+		IOMUX_PAD(0x0638, 0x0250, 2 | IOMUX_CONFIG_SION, 0x0860, 1,            \
+				MX6DL_I2C_PAD_CTRL)
 #define MX6DL_PAD_KEY_COL3__KPP_COL_3                                          \
 		IOMUX_PAD(0x0638, 0x0250, 3, 0x0000, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_COL3__I2C2_SCL                                           \
-		IOMUX_PAD(0x0638, 0x0250, 4 | IOMUX_CONFIG_SION, 0x0870, 1, MX6DL_I2C_PAD_CTRL)
+		IOMUX_PAD(0x0638, 0x0250, 4 | IOMUX_CONFIG_SION, 0x0870, 1,            \
+				MX6DL_I2C_PAD_CTRL)
 #define MX6DL_PAD_KEY_COL3__GPIO_4_12                                          \
 		IOMUX_PAD(0x0638, 0x0250, 5, 0x0000, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_COL3__SPDIF_IN1                                          \
@@ -2756,11 +2758,13 @@
 #define MX6DL_PAD_KEY_ROW3__ASRC_ASRC_EXT_CLK                                  \
 		IOMUX_PAD(0x064C, 0x0264, 1, 0x0794, 2, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_ROW3__HDMI_TX_DDC_SDA                                    \
-		IOMUX_PAD(0x064C, 0x0264, 2, 0x0864, 1, NO_PAD_CTRL)
+		IOMUX_PAD(0x064C, 0x0264, 2 | IOMUX_CONFIG_SION, 0x0864, 1,            \
+				MX6DL_I2C_PAD_CTRL)
 #define MX6DL_PAD_KEY_ROW3__KPP_ROW_3                                          \
 		IOMUX_PAD(0x064C, 0x0264, 3, 0x0000, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_ROW3__I2C2_SDA                                           \
-		IOMUX_PAD(0x064C, 0x0264, 4 | IOMUX_CONFIG_SION, 0x0874, 1, MX6DL_I2C_PAD_CTRL)
+		IOMUX_PAD(0x064C, 0x0264, 4 | IOMUX_CONFIG_SION, 0x0874, 1,            \
+				MX6DL_I2C_PAD_CTRL)
 #define MX6DL_PAD_KEY_ROW3__GPIO_4_13                                          \
 		IOMUX_PAD(0x064C, 0x0264, 5, 0x0000, 0, NO_PAD_CTRL)
 #define MX6DL_PAD_KEY_ROW3__USDHC1_VSELECT                                     \
diff --git a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
index 98e4471..7618975 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
@@ -2184,7 +2184,7 @@
 #define _MX6Q_PAD_KEY_COL3__ENET_CRS			\
 		IOMUX_PAD(0x05E0, 0x0210, 1, 0x0000, 0, 0)
 #define _MX6Q_PAD_KEY_COL3__HDMI_TX_DDC_SCL			\
-		IOMUX_PAD(0x05E0, 0x0210, 2, 0x0890, 1, 0)
+		IOMUX_PAD(0x05E0, 0x0210, 2 | IOMUX_CONFIG_SION, 0x0890, 1, 0)
 #define _MX6Q_PAD_KEY_COL3__KPP_COL_3			\
 		IOMUX_PAD(0x05E0, 0x0210, 3, 0x0000, 0, 0)
 #define _MX6Q_PAD_KEY_COL3__I2C2_SCL			\
@@ -2201,7 +2201,7 @@
 #define _MX6Q_PAD_KEY_ROW3__ASRC_ASRC_EXT_CLK		\
 		IOMUX_PAD(0x05E4, 0x0214, 1, 0x07B0, 0, 0)
 #define _MX6Q_PAD_KEY_ROW3__HDMI_TX_DDC_SDA			\
-		IOMUX_PAD(0x05E4, 0x0214, 2, 0x0894, 1, 0)
+		IOMUX_PAD(0x05E4, 0x0214, 2 | IOMUX_CONFIG_SION, 0x0894, 1, 0)
 #define _MX6Q_PAD_KEY_ROW3__KPP_ROW_3			\
 		IOMUX_PAD(0x05E4, 0x0214, 3, 0x0000, 0, 0)
 #define _MX6Q_PAD_KEY_ROW3__I2C2_SDA			\
@@ -5791,7 +5791,7 @@
 #define  MX6Q_PAD_KEY_COL3__ENET_CRS		\
 		(_MX6Q_PAD_KEY_COL3__ENET_CRS | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_KEY_COL3__HDMI_TX_DDC_SCL		\
-		(_MX6Q_PAD_KEY_COL3__HDMI_TX_DDC_SCL | MUX_PAD_CTRL(NO_PAD_CTRL))
+		(_MX6Q_PAD_KEY_COL3__HDMI_TX_DDC_SCL | MUX_PAD_CTRL(MX6Q_I2C_PAD_CTRL))
 #define  MX6Q_PAD_KEY_COL3__KPP_COL_3		\
 		(_MX6Q_PAD_KEY_COL3__KPP_COL_3 | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_KEY_COL3__I2C2_SCL		\
@@ -5808,7 +5808,7 @@
 #define  MX6Q_PAD_KEY_ROW3__ASRC_ASRC_EXT_CLK		\
 		(_MX6Q_PAD_KEY_ROW3__ASRC_ASRC_EXT_CLK | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_KEY_ROW3__HDMI_TX_DDC_SDA		\
-		(_MX6Q_PAD_KEY_ROW3__HDMI_TX_DDC_SDA | MUX_PAD_CTRL(NO_PAD_CTRL))
+		(_MX6Q_PAD_KEY_ROW3__HDMI_TX_DDC_SDA | MUX_PAD_CTRL(MX6Q_I2C_PAD_CTRL))
 #define  MX6Q_PAD_KEY_ROW3__KPP_ROW_3		\
 		(_MX6Q_PAD_KEY_ROW3__KPP_ROW_3 | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_KEY_ROW3__I2C2_SDA		\
-- 
1.7.9.5

